packages/langium/src/workspace/document-builder.ts

```diff
@@ -59,7 +59,7 @@ export interface DocumentBuilder {
 }
 
 export type DocumentUpdateListener = (changed: URI[], deleted: URI[]) => void
-export type DocumentBuildListener = (built: LangiumDocument[], cancelToken: CancellationToken) => Promise<void>
+export type DocumentBuildListener = (built: LangiumDocument[], cancelToken: CancellationToken) => void | Promise<void>
```

Comment:
Is `void | ` really needed? It is used as callback, which you normally never await.

Reply:
Or differently: Are you now not forced to check at every usage if it returns a Promise or not?

Reply:
No, you can just `await` the result regardless whether it returns a `Promise` or `undefined`. If you call `await` with something that is not `Thenable`, it just returns that value immediately.

Reply:
Oh ok, thanks for filling this knowledge gap :D